service-map/src/main/java/fi/nls/oskari/map/data/domain/GFIRequestParams.java

```diff
@@ -140,11 +146,21 @@ private String getAsQueryString() {
         }
 
 
-        return WMS_GFI_BASE_PARAMS + "&SRS=" + getSRSName() + "&BBOX=" + getBbox() + "&X=" + getX()
+        String query = WMS_GFI_BASE_PARAMS + "&SRS=" + getSRSName() + "&BBOX=" + getBbox() + "&X=" + getX()
                 + "&Y=" + getY() + "&INFO_FORMAT=" + getInfoFormat()
                 + "&QUERY_LAYERS=" + wmsName + "&WIDTH="
                 + getWidth() + "&HEIGHT=" + getHeight() + "&STYLES="
                 + getCurrentStyle() + "&LAYERS=" + wmsName;
+
+        if (additionalParams != null) {
+            Iterator<String> keys = additionalParams.keys();
+            while (keys.hasNext()) {
+                String key = keys.next();
+                String value = additionalParams.optString(key);
+                query += "&" + key.toUpperCase() + "=" + value;
+            }
+        }
```

Comment:
Use a method in IOHelper instead that will take care of urlencoding the param values etc.